core/block_dragger.js

```diff
@@ -0,0 +1,239 @@
+
+/**
+ * Start dragging a block.  This includes moving it to the drag surface.
+ */
+Blockly.BlockDragger.prototype.startBlockDrag = function() {
+  // Workspace.startDrag just records the start position, which we already know.
+  // TODO: Figure out where to get the list of bubbles to move when dragging.
+  // TODO: Can setResizesEnabled be done at the same time for both types of drags?
+  this.workspace_.setResizesEnabled(false);
+  // TODO: Add getParent() to blockSvg.
+  if (this.draggingBlock_.parentBlock_) {
+    this.draggingBlock_.unplug();
+    this.draggingBlock_.disconnectUiEffect();
+  }
+  // TODO: Make setDragging_ package.
+  this.draggingBlock_.setDragging_(true);
+  // TODO: Consider where moveToDragSurface should live.
+  this.draggingBlock_.moveToDragSurface_();
+  Blockly.Css.setCursor(Blockly.Css.Cursor.CLOSED);
+};
+
+/**
+ * Execute a step of block dragging, based on the given event.  Update the
+ * display accordingly.
+ * @param {!Event} e The most recent move event.
+ * @param {!goog.math.Coordinate} currentDragDeltaXY How far the pointer has
+ *     moved from the position at the start of the drag.
+ */
+Blockly.BlockDragger.prototype.dragBlock = function(e, currentDragDeltaXY) {
+  if (this.useDragSurface_) {
+    var newLoc = goog.math.Coordinate.sum(this.blockRelativeToSurfaceXY_,
+        currentDragDeltaXY);
+    this.dragSurface_.translateSurface(newLoc.x, newLoc.y);
+  }
+  this.deleteArea_ = this.workspace_.isDeleteArea(e);
+  // TODO: Handle the case when we aren't using the drag surface.
+  this.draggedConnectionManager_.update(currentDragDeltaXY, this.deleteArea_);
+  this.updateCursorDuringBlockDrag_();
+};
+
+/**
+ * Finish a block drag and put the block back on the workspace.
+ * @param {!Event} e The mouseup/touchend event.
+ * @param {!goog.math.Coordinate} currentDragDeltaXY How far the pointer has
+ *     moved from the position at the start of the drag.
+ */
+Blockly.BlockDragger.prototype.endBlockDrag = function(e, currentDragDeltaXY) {
+  // Make sure internal state is fresh.
+  this.dragBlock(e, currentDragDeltaXY);
+  Blockly.BlockSvg.disconnectUiStop_();
+  // TODO: Consider where moveOffDragSurface should live.
+  this.draggingBlock_.moveConnections_(currentDragDeltaXY.x,
+      currentDragDeltaXY.y);
+  this.draggingBlock_.moveOffDragSurface_();
+  var deleted = this.maybeDeleteBlock_();
+  if (!deleted) {
+    // setDragging_ is expensive and doesn't need to be done if we're deleting.
+    this.draggingBlock_.setDragging_(false);
+    this.draggedConnectionManager_.applyConnections();
+    this.draggingBlock_.render();
+    this.fireMoveEvent_();
+  }
+  this.workspace_.setResizesEnabled(true);
+  Blockly.Css.setCursor(Blockly.Css.Cursor.OPEN);
+};
+
+/**
+ * Fire a move event at the end of a block drag.
+ * @private
+ */
+Blockly.BlockDragger.prototype.fireMoveEvent_ = function() {
+  var event = new Blockly.Events.Move(this.draggingBlock_);
+  event.oldCoordinate = this.startCoordinate_;
```

Comment:
Hm.  Based on the contents of BlockSvg.moveBy, I should be using blockRelativeToSurfaceXY_.  @picklesrus, does that sound reasonable?

In that case I wouldn't need startCoordinate_ at all.